### DISCUSSION TITLE
Setup streaming replication

### DISCUSSION THREAD

**Discussion by ghost** [supabase/supabase#18211] [Questions]
Hi,

Anyone that have been successful setting up a streaming replication from Supabase database (primary database) to another Postgres database (standby database) not necessarily hosted by Supabase. All guides, [example](https://www.postgresql.fastware.com/postgresql-insider-ha-str-rep), I've been trying to follow instructs you to change pg_hba.conf file to accept incoming requests from the standby database, which doesn't seem possible?

Logs on my on-premises standby server indicated that I need to configure pg_hba.conf file on the primary server:
`FATAL:  no pg_hba.conf entry for replication connection from host "<IP-Number>", user "postgres", no encryption`

Supabase pg_hba.conf looks like this:
```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     trust
# IPv4 local connections:
host    all             all             127.0.0.1/32            trust
# IPv6 local connections:
host    all             all             ::1/128                 trust
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     trust
host    replication     all             127.0.0.1/32            trust
host    replication     all             ::1/128                 trust
```

What puzzles me is that I've tried the [Hevo](https://hevodata.com/) platform which I was able to setup a (streaming?) replication to another database. If it was using streaming replication or any other solution I'm not aware of but the changes showed up pretty much instant at the standby server. Unfortunately Hevo is not an option due to the hefty pricing tier.

Background for my question is that I'm building a system for a small association which registers money transactions for a flea market and this redundancy is necessary in case of a disaster or outage, not a single transaction can be lost.

**Comment by BoscoDomingo** [comment]
Hey @dea82 did you manage to get this working? All the tools I've tried thus far have failed, and I can't find any official support on this matter

**Reply by ghost** [reply]
Hi @BoscoDomingo,

No, but I've found a decent solution for my requirements. I've set up a [logical replication](https://www.postgresql.org/docs/current/logical-replication.html) instead. I'm not 100% percent sure what the difference behind the scene is because the changes in main database show up pretty much instant in the backup database so for me that's good enough.

Create a publication in the main database (in this case for all tables in a specific schema):
```sql
create publication <NAME_OF_PUBLICATION> for tables in schema <SCHEMA_NAME>; 
```

Create a subscription in the backup database.
```sql
create subscription <NAME_OF_SUBSCRIPTION> connection '<CONNECTION_STRING>' publication <NAME_OF_PUBLICATION>;
```
Where `<CONNECTION_STRING>` should look something like:
`host=db.somethingwhatever.supabase.co port=5432 dbname=postgres password=somesecret`

Hopefully this will be ok for your requirements as well?

**Reply by BoscoDomingo** [reply]
I'll try this in BigQuery (the destination for my data) but I'm not sure it is supported unfortunately. This should serve for a pure read replica though. Thanks for the help!!
